mozilla-central/setup

```diff
@@ -46,7 +49,102 @@ pushd $GIT_ROOT
 git pull
 popd
 
+date
+
+# Edit this to e.g. try.revision.ee64db93dcc149da9313460317257b8c42eec5b2 or whatever
+# to test other revisions. If you do this, also override INDEXED_GIT_REV below to some sane
+# git-equivalent of the revision because most likely it won't find anything in the mapfile.
+# This defaults to the most recent searchfox indexing run for mozilla-central.
+REVISION=mozilla-central.latest
+
+pushd $INDEX_ROOT
+wget -nv https://index.taskcluster.net/v1/task/gecko.v2.$REVISION.firefox.linux64-searchfox-debug/artifacts/public/build/target.json
+INDEXED_HG_REV=$(python $MOZSEARCH_PATH/scripts/read-json.py target.json moz_source_stamp)
+INDEXED_GIT_REV=$(awk -v REV=$INDEXED_HG_REV '$2 ~ REV { print $1 }' git_hg.map)
+
+# If INDEXED_GIT_REV gets set, that means the gecko-dev repo includes the code for which
+# The taskcluster indexing job ran. So we can download the result of that indexing
+# job and use it. Otherwise, it means gecko-dev is lagging behind the canonical hg
```

Comment:
It sounds like you are saying there is a race between the hg-to-git mirror gecko-dev and the SearchFox taskcluster build. If the taskcluster build wins the race, it sounds like we fall back to the current behavior of doing a "local" clang build? Should there be some output in the log indicating which thing happened, or is it clear enough from all of the other log spew from the various steps? How often do we expect to fall back to the non-taskcluster build? Do we have any way of knowing if it starts happening very frequently? It seems like that would be bad (though not bad enough that we'd want SearchFox to break entirely instead). If we're only rarely using the "local" build path, how are we going to ensure that it keeps working?

Reply:
> It sounds like you are saying there is a race between the hg-to-git mirror gecko-dev and the SearchFox taskcluster build. If the taskcluster build wins the race, it sounds like we fall back to the current behavior of doing a "local" clang build?

Yes.

> Should there be some output in the log indicating which thing happened, or is it clear enough from all of the other log spew from the various steps?

It will be clear from the log spew, but I do want to add a "warning" mechanism where it will send an email to the mailing list (but still complete the indexing successfully) in this case and other similar cases. I'm dling that separately though.

>  How often do we expect to fall back to the non-taskcluster build?

Not very often. These days the git mirror is pretty good at staying updated.

> Do we have any way of knowing if it starts happening very frequently?

With just this patch you'd have to manually check the log. With the warning mechanism I described above we'd get notified by email every time it happens so it would be easy to keep an eye out for it.

> If we're only rarely using the "local" build path, how are we going to ensure that it keeps working?

This is a potential problem, yes. But I'm hoping that the set of build dependencies doesn't expand that often that it's a major concern. One thing I want to do that will help here is to run `mach bootstrap` before `mach build` in the `build` script which should do a better job of ensuring the "local" build path doesn't break.